In [1]:
API_KEY = "23a44102e1b21d8bc08149ae9622078b"

In [2]:
import requests

API_KEY = "23a44102e1b21d8bc08149ae9622078b"

url = f"https://api.openweathermap.org/data/2.5/weather?q=Lagos&appid={API_KEY}&units=metric"

response = requests.get(url)

print(response.status_code)
print(response.json())

200
{'coord': {'lon': 3.75, 'lat': 6.5833}, 'weather': [{'id': 804, 'main': 'Clouds', 'description': 'overcast clouds', 'icon': '04d'}], 'base': 'stations', 'main': {'temp': 28.76, 'feels_like': 31.87, 'temp_min': 28.76, 'temp_max': 28.76, 'pressure': 1012, 'humidity': 68, 'sea_level': 1012, 'grnd_level': 1012}, 'visibility': 10000, 'wind': {'speed': 4.35, 'deg': 230, 'gust': 5.35}, 'clouds': {'all': 97}, 'dt': 1786890250, 'sys': {'country': 'NG', 'sunrise': 1786858772, 'sunset': 1786903163}, 'timezone': 3600, 'id': 2332453, 'name': 'Lagos', 'cod': 200}


In [3]:
import requests
import pandas as pd
API_KEY =  "23a44102e1b21d8bc08149ae9622078b"
cities = ["Lagos", "Abuja", "Enugu"]
weather_data = []
for city in cities:
    url =  f"https://api.openweathermap.org/data/2.5/weather?q={city}&appid={API_KEY}&units=metric"
    response = requests.get(url)
    if response.status_code == 200:
        data = response.json()
        weather_data.append({
            "city":city, "Temperature (°C)":
            data["main"]["temp"], "Humidity (%)":
            data["main"]["humidity"], "weather condidtion":
            data["weather"][0]["description"], "Wind Speed (m/s)":
            data["wind"]["speed"], "Date and Time": data["dt"]})
        df = pd.DataFrame(weather_data)
        print(df)

    city  Temperature (°C)  Humidity (%) weather condidtion  Wind Speed (m/s)  \
0  Lagos             28.76            68    overcast clouds              4.35   

   Date and Time  
0     1786890250  
    city  Temperature (°C)  Humidity (%) weather condidtion  Wind Speed (m/s)  \
0  Lagos             28.76            68    overcast clouds              4.35   
1  Abuja             28.00            65    overcast clouds              1.37   

   Date and Time  
0     1786890250  
1     1786890619  
    city  Temperature (°C)  Humidity (%) weather condidtion  Wind Speed (m/s)  \
0  Lagos             28.76            68    overcast clouds              4.35   
1  Abuja             28.00            65    overcast clouds              1.37   
2  Enugu             27.03            81      moderate rain              1.29   

   Date and Time  
0     1786890250  
1     1786890619  
2     1786890901  


In [4]:
df["Date and Time"] = pd.to_datetime(
    df["Date and Time"],
    unit="s")

In [5]:
df.rename(columns={"weather condidtion": "Weather Condition"},inplace=True)

In [6]:
df.rename(columns={"city": "City"},inplace=True)

In [7]:
print(df.dtypes)

City                           str
Temperature (°C)           float64
Humidity (%)                 int64
Weather Condition              str
Wind Speed (m/s)           float64
Date and Time        datetime64[s]
dtype: object


In [9]:
print(df.isnull().sum())
print(df.duplicated().sum())

City                 0
Temperature (°C)     0
Humidity (%)         0
Weather Condition    0
Wind Speed (m/s)     0
Date and Time        0
dtype: int64
0


In [10]:
print(df)

    City  Temperature (°C)  Humidity (%) Weather Condition  Wind Speed (m/s)  \
0  Lagos             28.76            68   overcast clouds              4.35   
1  Abuja             28.00            65   overcast clouds              1.37   
2  Enugu             27.03            81     moderate rain              1.29   

        Date and Time  
0 2026-08-16 14:24:10  
1 2026-08-16 14:30:19  
2 2026-08-16 14:35:01  


In [11]:
df.to_csv("weather_data.csv", index=False)

In [12]:
df.to_excel("weather_data.xlsx", index=False)

In [13]:
import sqlite3
conn = sqlite3.connect("weather.db")
df.to_sql("weather_data",conn,if_exists="replace",index
          =False)
conn.close()

## COMPARING TEMPERATURES ACROSS CITIES

In [14]:
print(df[["City", "Temperature (°C)"]])

    City  Temperature (°C)
0  Lagos             28.76
1  Abuja             28.00
2  Enugu             27.03


## CITY WITH HIGHEST HUMIDITY

In [16]:
Highest_Humidity = df.loc[df["Humidity (%)"].idxmax()]
print("City with Highest Humidity:", Highest_Humidity["City"])

City with Highest Humidity: Enugu


## COMPARING WEATHER CONDITIONS

In [17]:
print(df[["City", "Weather Condition"]])

    City Weather Condition
0  Lagos   overcast clouds
1  Abuja   overcast clouds
2  Enugu     moderate rain


## RESULT AND FINDING

Weather data collected on 18 August 2026 shows differences in temperature, humidity and weather conditions in Lagos, Abuja and Enugu.
Lagos was 28.76 degree Celsius with humidity level of 68%. The weather condition was overcast clouds meaning the sky was mostly covered with clouds which means less direct sunlight and a cloudy atmosphere in general.
In Abuja the temperature was 28.00°C and the humidity was 65%. Like Lagos, Abuja also had overcast clouds which meant that the sky was mostly covered by clouds and there was little or no direct sunshine.
Enugu recorded a temperature of 27.03°C and the highest humidity level of 81 per cent. Weather condition was moderate rain that is continuous rain of moderate intensity. This increased humidity is consistent with the rain in the city.
Lagos had the highest temperature at 28.76°C and Enugu had the highest humidity at 81%. The moderate rain in Enugu contributes to its higher humidity, while overcast conditions in Lagos and Abuja suggest cloudy weather without substantial rain.